src/components/UI/Modal/PlayListViewModal/PlayListViewModal.jsx

```diff
@@ -0,0 +1,65 @@
+import axios from 'axios';
+import React from 'react'
+import { usePlayList } from '../../../../context/PlayListContext';
+import "./PlayListViewModal.css";
+
+
+function PlayListViewModal({ data, setPlayListModal}) {
+    const { playListContextArray, setPlayListContextArray } = usePlayList();
+    const { videoDetails } = data;
+    console.log(videoDetails, data,setPlayListModal);
+
+
+    /**
+     * 
+     * @param {string} name 
+     * method is to add a video in playlist
+     */
+    const SelectedPlayListHandler = async (playlistItem) => {
+        console.log(playlistItem);
+        // /user/playlists/:playlistId
+        try {
+            var res = await axios.post(`/api/user/playlists/${playlistItem}`,
+                { "video": { ...data } },
+                {
+                    headers: { authorization: localStorage.getItem("jafnaToken") }
+                });
+            console.log(res);
+            // setPlayListContextArray(res.data.playlist);
+            try{
+                (async() => {
+                var res = await axios.get("/api/user/playlists",{ headers:{
+                  authorization:localStorage.getItem("jafnaToken")}});
+                  console.log(res);
+                  setPlayListContextArray(res.data.playlists);
+                //   setPlaylist(res.data.playlists);
+            })()}
+            catch(error){
+              console.log("Product list page error",error);
+            }
+        } catch (err) {
```

Comment:
While doing the post request you are getting some responses. You can also use that response instead of doing another API for it.

Reply:
actually, the post request only returns that playlist data for which we did post request and this API call will get us all the playlist data again

Reply:
but I guess I wont require the whole playlist data to be updated in playlist context as on playlist render I am fetching a get request and get all playlist data again
thanks for the comment made me realize I don't need the API call